Mathlib/Geometry/Euclidean/Similarity.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2025 Chu Zheng. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chu Zheng
+-/
+module
+
+import Mathlib.Geometry.Euclidean.Triangle
+import Mathlib.Topology.MetricSpace.Similarity
+
+/-!
+# Triangle Similarity
+
+This file contains theorems about similarity of triangles, including conditions
+for similarity based on sides and angles.
+
+-/
+
+open scoped Congruent EuclideanGeometry
+
+open Similar NNReal Affine
+
+namespace EuclideanGeometry
+
+variable {ι V₁ V₂ P₁ P₂ : Type*}
+  [NormedAddCommGroup V₁] [NormedAddCommGroup V₂]
+  [InnerProductSpace ℝ V₁] [InnerProductSpace ℝ V₂]
+  [MetricSpace P₁] [MetricSpace P₂]
+  [NormedAddTorsor V₁ P₁] [NormedAddTorsor V₂ P₂]
+  {v₁ : ι → P₁} {v₂ : ι → P₂}
+  {a b c : P₁} {a' b' c' : P₂}
+
+/-- If two triangles have two pairs equal angles, then the triangles are similar. -/
+theorem similar_of_angle_angle (h_not_col : ¬ Collinear ℝ {a, b, c})
+    (h_not_col' : ¬ Collinear ℝ {a', b', c'})
+    (h₁ : ∠ a b c = ∠ a' b' c')
+    (h₂ : ∠ b c a = ∠ b' c' a') :
+    ![a, b, c] ∼ ![a', b', c'] := by
```

Comment:
Do you actually need both "not collinear" hypotheses? I think you should only need the first one; that one implies that none of the angles between `a`, `b` and `c` are 0 or pi, so the only angle consistent with collinearity of the second triple of points is pi/2 (consistent with collinearity in the case of an angle where at least one vector is 0), but it's not possible for the two given angles among non-collinear `a`, `b`, `c` to both be pi/2.